wellregistry/registry/admin/monitoring_location.py

```diff
@@ -217,10 +220,100 @@ def _has_permission(perm, user, obj=None):
     return user.has_perm(perm) and (not obj or obj.agency.agency_cd in _get_groups(user))
 
 
+def _get_usergroups(request):
+    user_groups = ""
+    if not request.user.is_superuser:
+        user_groups = re.search(
+            "\[(.*)\]", str(_get_groups(request.user))).group(1)
+        user_groups = " where agency.agency_cd IN (" + user_groups + ") "
+    return user_groups
+
+
+def _get_where_clause(request, user_groups):
+    where_clause = ""
+    sep = ""
+    query_params = request.body.decode().split("&")
+    for query_param in query_params:
+        param = query_param.split("=")
+        if param[0] == "_selected_action":
+            where_clause += sep + param[1]
+            sep = ","
```

Comment:
thinking about it, I think python has a join method that might be better.
https://www.programiz.com/python-programming/methods/string/join

Reply:
i will look into it